xbmc/music/tags/TagLoaderTagLib.cpp

```diff
@@ -565,6 +565,28 @@ bool CTagLoaderTagLib::ParseTag(APE::Tag *ape, EmbeddedArt *art, CMusicInfoTag&
       tag.SetMusicBrainzTrackID(it->second.toString().to8Bit(true));
     else if (it->first == "MUSICBRAINZ_ALBUMTYPE")
       SetReleaseType(tag, StringListToVectorString(it->second.toStringList()));
+    else if (it->first == "COVER ART (FRONT)")
+    {
+      TagLib::ByteVector tdata = it->second.binaryData();
+      // The image data follows a null byte, which can optionally be preceded by a filename
+      const uint offset = tdata.find('\0') + 1;
```

Comment:
Please keep broken and ill crafted files in mind.

Reply:
Any suggestions as to furthering this aim here? The decoding libraries are much more aware of invalid bitstreams than we can be here.

> Please keep broken and ill crafted files in mind.

Do you have any suggestions for further addressing this here? The decoding libraries are appropriately much more responsible for the handling of invalid bitstreams.

Reply:
the find call returns -1 if nothing was found, which yields the entire data in the buffer. which is the proper thing to do. this is fine.

Reply:
I am afraid if it is the last byte and that was why I wrote the comment.

Reply:
The offset has to be less than the size of the data and greater than 0, requiring that the data size is > 1. If the null is the last byte, the resulting bv data size would be less than 0.

The offset could be required <= the data size (thus implying a required data size > 0).

The mime checks require a data size of !< 2, however it does look more correct to allow the offset to equal but never overtake the data size.